error.go

```diff
@@ -36,6 +46,52 @@ func (e internalError) InternalError() error {
 	return e.error
 }
 
+// SetCode set the error's translation code.
+func (e Error) SetCode(code string) Error {
```

Comment:
For setters, we better use pointer receivers as it is more common and can avoid mistakes (e.g. calling the setter without receiving the result)

Reply:
Yea, In my sources I usually define most methods by pointer (by my need), but here because I saw you do not get a pointer, I just coded like your sources.